src/common/state/table.c

```diff
@@ -101,8 +96,11 @@ int64_t table_timeout_handler(event_loop *loop,
     return EVENT_LOOP_TIMER_DONE;
   }
 
-  /* Decrement retry count and try again. */
-  callback_data->retry.num_retries--;
+  /* Decrement retry count and try again. We use -1 to indicate infinite
+   * retries. */
+  if (callback_data->retry.num_retries != -1) {
```

Comment:
If I try it on the workload here https://github.com/ray-project/ray/issues/115#issuecomment-282136339, then I see the logging message

```
[WARN] (/Users/rkn/Workspace/ray/src/common/state/table.c:84) retrying operation, retry_count = -1
```

printed a bunch of times. And the workload seems to work. In the tests, I think the most likely place for this codepath to get triggered is during the stress tests.